internal/scheduleringester/schedulerdb.go

```diff
@@ -242,12 +258,32 @@ func (s *SchedulerDb) WriteDbOp(ctx *armadacontext.Context, tx pgx.Tx, op DbOper
 		if err != nil {
 			return errors.WithStack(err)
 		}
+		for runId, failInfo := range o {
+			params := schedulerdb.SetTerminatedTimeParams{
+				RunID:               runId,
+				TerminatedTimestamp: &failInfo.FailureTime,
+			}
+			err = queries.SetTerminatedTime(ctx, params)
+			if err != nil {
+				return errors.WithStack(err)
+			}
+		}
 	case MarkRunsRunning:
 		runIds := maps.Keys(o)
 		err := queries.MarkJobRunsRunningById(ctx, runIds)
```

Comment:
Upserting many rows in one call is much more efficient than making one call per job. There should be some way to create a function
```
queries.MarkJobRunsRunningById(ctx, runIds, timestamps)
```
Or
```
queries.MarkJobRunsRunningById(ctx, map[runIds]timestamps)
```
Of these, the first is probably easier to write in sql.

Reply:
I made it to be `queries.MarkJobRunsRunningById(ctx, runIds, timestamp)` , the function will receive a single timestamp object, couldn't completely figure out how to make sqlc update multiple rows with an array of values. However, the `JobRunsRunning` instruction will have a single `timestamp` value, not multiple times, so we can update the job status with this single value provided ( The way it is after my last commit)